taichi/backends/opengl/codegen_opengl.cpp

```diff
@@ -848,11 +848,17 @@ class KernelGen : public IRVisitor {
   void visit(ReturnStmt *stmt) override {
     used.buf_args = true;
     // TODO: use stmt->ret_id instead of 0 as index
-    emit("_args_{}_[{} >> {} + 0] = {};",
-         opengl_data_type_short_name(stmt->element_types()[0]),
-         taichi_opengl_ret_base,
-         opengl_data_address_shifter(stmt->element_types()[0]),
-         stmt->values[0]->short_name());
+    int idx{0};
+    for (auto &value : stmt->values) {
+      emit("_args_{}_[({} >> {}) + {}] = {};",
+           opengl_data_type_short_name(value->element_type()),
```

Comment:
Is the `element_type()` here guaranteed to be `i32`?

Reply:
fix